mbotmake2/grammars/thumbnails.py

```diff
@@ -10,13 +10,16 @@
 BlankLine = ~r"[ \t\r]*\n"i
 Timestamp = "; generated by " ~r"[^\n]+"i newline
 
-Header = ";\n; thumbnail begin " Integer "x" Integer " " Integer newline
+Header = ORCA_TAG? ";\n; thumbnail begin " Integer "x" Integer " " Integer newline
 Payload = Colon ~r"[^\n ]+"i newline
-Footer = "; thumbnail end\n;\n"
+Footer = "; thumbnail end\n" (ORCA_TAG / BlankComment)
 
 Comment = ~r";[^\n]*\n"
 Command = ~r"[GM]\d+[^\n]*\n"
 
+ORCA_TAG = ~r"; THUMBNAIL_BLOCK_[^\n]+" newline+
+BlankComment = ";\n"
+
```

Comment:
In plain English: the header contains a line `; thumbnail begin 1234x256 1234`, but sometimes, other slicers append an extra line such as `; THUMBNAIL_BLOCK_****`. If exists, capture it as part of the header, and then ignore it.

Similarly, the footer `; thumbnail end` sometimes comes with an additional line `; THUMBNAIL_BLOCK_***`. If we encounter this optional line, capture it as part of the footer, and then discard it.